assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -870,8 +878,32 @@ describe("DiversionPage", () => {
     })
 
     await waitFor(() => {
-      expect(originalRouteShape.get(container)).toBeVisible()
+      expect(screen.queryByTitle("Detour Start")).toBeVisible()
     })
+
+    expect(originalRouteShape.get(container)).toBeVisible()
+  })
+
+  test("replaces the original route shape with an unfinished segment after the start point is added if the user is in the right test group", async () => {
+    jest
```

Comment:
This test (and the one above it) is (are?) a bit devious. 

Because route segments show up in response to an API call, the test above (the one that makes sure that `originalRouteShape` is still there when the test group isn't enabled) still passes, even if we don't gate it by test group, because the `fetch` call defaults to `neverPromise()`. 

But if I set the `fetch` call to default to having a realistic return value, then a whole bunch of the tests start complaining that there's a change not wrapped in `act(...)`, because the promise resolves after the test is finished executing.

I'm not sure how to get around this 🙃

Reply:
> But if I set the fetch call to default to having a realistic return value, then a whole bunch of the tests start complaining

If you mock the `fetch` call right in the test test `keeps the original route shape after...`, I don't see why other tests would start complaining, so I think I'm missing something simple 🤔

Reply:
Oh - if I mock it only in that test, then that test fails because the `fetch` resolves after the test wraps up, and I don't know how to make the test wait for the `fetch` to resolve, since the `fetch` resolving doesn't cause any changes on the page....

At least I _think_ that's what's happening here?

Reply:
Turns out [this](https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#an-alternative-waiting-for-the-mocked-promise) works! So we can wait for the promise returned by `fetchUnfinishedDetour` to resolve and _then_ assert that the route segments on the screen are correct, then that test tests what we want (and fails if I don't do the test group correctly)